libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,479 @@
+ * file, folder, synbolic link, hard link, directory junction, etc.
+ *
+ * @public
+ */
+export class FileSystem {
+
+  // ===============
+  // COMMON OPERATIONS
+  // ===============
+
+  /**
+   * Returns true if the path exists on disk.
+   * Behind the scenes it uses `fs.existsSync()`.
+   * @remarks
+   * There is a debate about the fact that after `fs.existsSync()` returns true,
+   * the file might be deleted before fs.readSync() is called, which would imply that everybody
+   * should catch a `readSync()` exception, and nobody should ever use `fs.existsSync()`.
+   * We find this to be unpersuasive, since "unexceptional exceptions" really hinder the
+   * break-on-exception debugging experience. Also, throwing/catching is generally slow.
+   * @param path - The absolute or relative path to the filesystem object.
+   */
+  public static exists(path: string): boolean {
+    return fsx.existsSync(path);
+  }
+
+  /**
+   * Gets the statistics for a particular filesystem object.
+   * If the path is a link, this function follows the link and returns statistics about the link target.
+   * Behind the scenes it uses `fs.statSync()`.
+   * @param path - The absolute or relative path to the filesystem object.
+   */
+  public static getStatistics(path: string): fs.Stats {
+    return fsx.statSync(path);
+  }
+
+  /**
+   * Updates the accessed and modified timestamps of the filesystem object referenced by path.
+   * Behind the scenes it uses `fs.utimesSync()`.
+   * The caller should specify both times in the `times` parameter.
+   * @param path - The path of the file that should be modified.
+   * @param times - The times that the object should be updated to reflect.
+   */
+  public static updateTimes(path: string, times: IUpdateTimeParameters): void {
+    // tslint:disable-next-line:no-any
+    fsx.utimes(path, times.accessedTime as any, times.modifiedTime as any);
+  }
+
+  /**
+   * Changes the permissions (i.e. file mode bits) for a filesystem object.
+   * Behind the scenes it uses `fs.chmodSync()`.
+   * @param path - The absolute or relative path to the object that should be updated.
+   * @param mode - UNIX-style file mode bits (e.g. 777 or 666 etc)
+   */
+  public static changePermissionBits(path: string, mode: IFileModeBits): void {
+    // tslint:disable-next-line:no-bitwise
+    const modeAsOctal: number = (mode.Owner << 6) + (mode.Group << 3) + (mode.Other);
+    fsx.chmodSync(path, modeAsOctal);
+  }
+
+  /**
+   * Moves a file. The folder must exist, unless the `ensureFolder` option is provided.
+   * Behind the scenes it uses `fsx.moveSync()`
+   * @param sourcePath - The absolute or relative path to the source file.
+   * @param targetPath - The absolute or relative path where the file should be moved to.
+   * @param options - Optional settings that can change the behavior. Type: `IFileSystemMoveOptions`
+   */
+  public static move(sourcePath: string, targetPath: string, options?: IFileSystemMoveOptions): void {
```

Comment:
>move [](start = 16, length = 4)

copyFile() and writeFile() both overwrite by default, and we found that this default is what 99% of our use cases want.

Logically I think move() should follow the same semantics.

Eventually all three of these should support throwIfExists=true, but that can be out of scope.  This replaces IFileSystemMoveOptions.overwrite.